itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -1719,4 +1746,85 @@ public Boolean apply(@Nullable CallerArguments args) {
             .run("select country from t2 order by country")
             .verifyResults(Arrays.asList("china", "india"));
   }
+
+  private void testReplicatedStatsForTable(String tableName) throws Exception {
+    // Test column stats
+    List<ColumnStatisticsObj> primaryColStats;
+    List<ColumnStatisticsObj> replicaColStats;
+    primaryColStats = primary.getTableColumnStatistics(primaryDbName, tableName);
+    replicaColStats = replica.getTableColumnStatistics(replicatedDbName, tableName);
+    LOG.debug("Column stats for table " + tableName + " on primary: " + primaryColStats);
+    LOG.debug("Column stats for table " + tableName + " on replica: " + replicaColStats);
+    Assert.assertEquals(primaryColStats, replicaColStats);
+
+    // Test table level stats
+    Map<String, String> pParams = primary.getTable(primaryDbName, tableName).getParameters();
+    Map<String, String> rParams = replica.getTable(replicatedDbName, tableName).getParameters();
+    LOG.debug("parameters of table " + tableName + " on primary " + pParams);
+    LOG.debug("parameters of table " + tableName + " on replica " + rParams);
+    for (String param : StatsSetupConst.TABLE_PARAMS_STATS_KEYS) {
+      LOG.debug("Checking statistics parameter " + param + " of table " + tableName + ". On " +
+              "primary: " + pParams.get(param) + " on secondary " + rParams.get(param));
+      Assert.assertEquals(pParams.get(param), rParams.get(param));
+    }
+  }
+
+  @Test
+  public void testStatsReplication() throws Throwable {
+    String simpleTableName = "sTable";
+    String partTableName = "pTable";
+    String ndTableName = "ndTable";
+
+    WarehouseInstance.Tuple bootstrapTuple = primary
+            .run("use " + primaryDbName)
+            .run("create table " + simpleTableName + " (id int)")
```

Comment:
For replicating stats to replica with strict.managed=true should include write-id in the table when update stats as managed table from primary will become ACID table at target and so it should map to a write_id for valid txn stats.

Reply:
Should we create a JIRA under https://issues.apache.org/jira/browse/HIVE-21037?

Reply:
nope. https://issues.apache.org/jira/browse/HIVE-20882.

Reply:
Done. https://issues.apache.org/jira/browse/HIVE-21108. Somehow I need to make sure that this gets worked upon before HIVE-21037 completes, mostly within HIVE-21078. For now I have linked HIVE-21108 to HIVE-21037.

Reply:
ok